save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/url/ParseUrlInTab.kt

```diff
@@ -0,0 +1,63 @@
+@file:Suppress("FILE_NAME_MATCH_CLASS")
+
+package com.saveourtool.save.frontend.components.views.url
+
+import com.saveourtool.save.domain.Role
+import com.saveourtool.save.entities.benchmarks.TabMenuBar
+import com.saveourtool.save.frontend.components.views.AbstractView
+import com.saveourtool.save.utils.URL_PATH_DELIMITER
+
+import react.State
+
+import kotlinx.browser.window
+
+/**
+ * General interface for working with MenuBar
+ */
+external interface HasSelectedMenu<T : Enum<T>> : State {
+    /**
+     * selected value in T Enum
+     */
+    var selectedMenu: T
+}
+
+/**
+ * The function of analyzing the URL of a tabbed page goes to the tab that was entered in the url, according to the role
+ *
+ * @param menu
+ * @param role
+ * @param isOrganizationCanCreateContest is state.organization?.canCreateContests in OrganizationView.kt
+ */
+fun <T, S : HasSelectedMenu<T>> AbstractView<*, S>.urlAnalysis(menu: TabMenuBar<T>, role: Role, isOrganizationCanCreateContest: Boolean?) {
+    val href = window.location.href
+    val tab = if (href.contains(menu.regexForUrlClassification)) {
+        href.substringAfterLast(URL_PATH_DELIMITER).let { menu.findEnumElement(it) ?: menu.defaultTab }
+    } else {
+        menu.defaultTab
+    }
+    if (state.selectedMenu != tab) {
+        if (menu.isNotAvailableWithThisRole(role, tab, isOrganizationCanCreateContest)) {
+            changeUrl(menu.defaultTab, menu)
+            window.alert("Your role is not suitable for opening this page")
```

Comment:
```suggestion
            window.alert("Your role is not suitable for opening this page")
            changeUrl(menu.defaultTab, menu)
```
With the current order the user will see the new link when the alert appears, may be confusing

Reply:
Done